actionmailer/lib/action_mailer/base.rb

```diff
@@ -900,15 +913,19 @@ def collect_responses(headers)
           yield(collector)
           collector.responses
         elsif headers[:body]
-          [{
-            body: headers.delete(:body),
-            content_type: self.class.default[:content_type] || "text/plain"
-          }]
+          collect_responses_from_text(headers)
         else
           collect_responses_from_templates(headers)
         end
       end
 
+      def collect_responses_from_text(headers)
+        [{
+          body: headers.delete(:body),
+          content_type: headers[:content_type] || self.class.default[:content_type] || "text/plain"
```

Comment:
If `:content_type` is passed to `mail`, `apply_defaults` will not copy the default value to the headers and `headers[:content_type]` will be the same as the `:content_type` passed to `mail`.

If `:content_type` is not passed to `mail`, `apply_defaults` will copy the default value to the headers and `headers[:content_type]` will be the same as the `self.class.default[:content_type`.

So at this point `headers[:content_type]` behaves exactly as we want.